kotlinx-collections-immutable/tests/src/test/kotlin/kotlinx.collections.immutable/stressTests/immutableList/PersistentListBuilderTest.kt

```diff
@@ -286,6 +287,180 @@ class PersistentListBuilderTest {
         }
     }
 
+    private fun <E> testIteratorsState(expected: ListIterator<E>, actual: ListIterator<E>) {
+        assertEquals(expected.hasPrevious(), actual.hasPrevious())
+        assertEquals(expected.previousIndex(), actual.previousIndex())
+        assertEquals(expected.hasNext(), actual.hasNext())
+        assertEquals(expected.nextIndex(), actual.nextIndex())
+    }
```

Comment:
Looks similar to `compare(expected, actual) { propertyEquals { ... } }` from other collection tests

Reply:
It turns out `compare(expected, actual) { listIteratorProperties() }` does exactly what is needed.